java/com/google/cose/Recipient.java

```diff
@@ -0,0 +1,121 @@
+package com.google.cose;
+
+import co.nstant.in.cbor.CborBuilder;
+import co.nstant.in.cbor.CborException;
+import co.nstant.in.cbor.builder.ArrayBuilder;
+import co.nstant.in.cbor.model.ByteString;
+import co.nstant.in.cbor.model.DataItem;
+import co.nstant.in.cbor.model.Map;
+import com.google.cose.exceptions.CoseException;
+import com.google.cose.utils.CborUtils;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implements COSE_Recipient structure.
+ */
+public class Recipient extends CoseMessage {
+  private final byte[] ciphertext;
+  private final List<Recipient> recipients;
+
+  private Recipient(byte[] protectedHeaderBytes, Map unprotectedHeaders, byte[] ciphertext,
+      List<Recipient> recipients) {
+    super(protectedHeaderBytes, unprotectedHeaders);
+    this.ciphertext = ciphertext;
+    this.recipients = recipients;
```

Comment:
Suggest ensuring `this.recipients` is never null, so callers don't have to worry about getting a null value when calling `getRecipients`. e.g. `this.recipients = recipients == null ? new List<>() : recipients`

Reply:
Recipients field in Recipient class is a nullable field actually since the cose spec mentions that it may or may not be present for this structure.